cmd/mount_zfs/Makefile.am

```diff
@@ -13,6 +13,6 @@ mount_zfs_SOURCES = \
 	mount_zfs.c
 
 mount_zfs_LDADD = \
-	$(top_builddir)/lib/libnvpair/libnvpair.la \
+	$(top_builddir)/lib/libzfs/libzfs.la \
 	$(top_builddir)/lib/libzfs_core/libzfs_core.la \
-	$(top_builddir)/lib/libzfs/libzfs.la
+	$(top_builddir)/lib/libnvpair/libnvpair.la
```

Comment:
At least with the libtool in my distro (gentoo), it is mostly for neatness. This is the way you'd have to link it when calling the linker directly, so that eg libnvpair.a can be used to satisfy dependencies from libzfs_core.a. However, libtool will add libnvpair when it sees libzfs_core because it knows about the dependency, and then remove the earlier libnvpair to reduce duplicates, so you end up with the correct order being passed to the linker anyway.

There seems to be something slightly different with the libtool in debian/ubuntu though, I saw build failures because not all libraries that the executable used directly were listed in LDADD before, even though they were dependencies of the listed libraries, and it didn't cause errors on my libtool version. This got exposed when I dropped libudev, libuutil and lib from $LIBS. It would be helpful I think to add V=1 LIBTOOLFLAGS="-v" to the buildbot make command lines to show exactly what it ends up doing.

Reply:
Got it, that makes sense.  We've definitely some slight variation with `libtool` across distributions, so I'm always a little concerned we'll hit something unexpected when adjusting the dependencies.  Thanks for the explanation.